src/Amalgam/GeneralizedDistance.h

```diff
@@ -748,7 +733,8 @@ class GeneralizedDistance
 		if(a_type == ENIVT_NULL || b_type == ENIVT_NULL)
 			return std::numeric_limits<double>::quiet_NaN();
 
-		if(feature_type == FDT_NOMINAL)
+		if(feature_type == FDT_NOMINAL_NUMERIC
+			|| feature_type == FDT_NOMINAL_STRING || feature_type == FDT_NOMINAL_CODE)
```

Comment:
I would add a new __forceinline IsFeatureNominal() for this like you did with the index.